packages/runtime/src/modules/DeciderModule.ts

```diff
@@ -22,18 +75,261 @@ export class DeciderModule extends RuntimeModule {
         if (event.data.newStatus !== LocalRequestStatus.DecisionRequired) return;
 
-        if (event.data.request.content.items.some(flaggedAsManualDecisionRequired)) return await this.requireManualDecision(event);
+        const requestContent = event.data.request.content;
+        if (this.containsItem(requestContent, (item) => item["requireManualDecision"] === true)) {
+            return await this.requireManualDecision(event);
+        }
+
+        const automationResult = await this.automaticallyDecideRequest(event);
+        if (automationResult.isSuccess) {
+            const services = await this.runtime.getServices(event.eventTargetAddress);
+            await this.publishEvent(event, services, "RequestAutomaticallyDecided");
+            return;
+        }
 
         return await this.requireManualDecision(event);
     }
 
+    public async automaticallyDecideRequest(event: IncomingRequestStatusChangedEvent): Promise<Result<LocalRequestDTO>> {
+        if (!this.configuration.automationConfig) return Result.fail(RuntimeErrors.deciderModule.doesNotHaveAutomationConfig());
+
+        const request = event.data.request;
+        const itemsOfRequest = request.content.items;
+
+        let decideRequestItemParameters = this.createResponseItemsWithSameDimension(itemsOfRequest, undefined);
+
+        for (const automationConfigElement of this.configuration.automationConfig) {
+            const requestConfigElement = automationConfigElement.requestConfig;
+            const responseConfigElement = automationConfigElement.responseConfig;
+
+            if (isGeneralRequestConfig(requestConfigElement)) {
+                const generalRequestIsCompatible = this.checkCompatibility(requestConfigElement, request);
+                if (!generalRequestIsCompatible) {
+                    continue;
+                }
+
+                const decideRequestItemParameterResult = this.createDecideRequestItemParametersForGeneralResponseConfig(event, responseConfigElement);
+                if (decideRequestItemParameterResult.isError) {
+                    continue;
+                }
+
+                const decideRequestResult = await this.decideRequest(event, decideRequestItemParameterResult.value);
+                return decideRequestResult;
+            }
+
+            if (isRequestItemDerivationConfig(requestConfigElement)) {
+                const checkCompatibilityResult = this.checkRequestItemCompatibilityAndApplyReponseConfig(
+                    itemsOfRequest,
+                    decideRequestItemParameters,
+                    request,
+                    requestConfigElement,
+                    responseConfigElement
+                );
+                if (checkCompatibilityResult.isError) {
+                    continue;
+                }
+
+                decideRequestItemParameters = checkCompatibilityResult.value;
+                if (!this.containsItem(decideRequestItemParameters, (element) => element === undefined)) {
```

Comment:
Could just be me, but it feels cleaner to separate the filling of the decideRequestItemParameters and the actual deciding

Reply:
this was my original idea (at the cost of performance) since it's then possible to break after processing a general request config. But I'd now rather move decideRequest out of the if-block but still keep it inside the for-block - and at the same time change the general request config logic to continue processing more configs after them (as stated in a different comment).